proxy/src/control/discovery.rs

```diff
@@ -187,7 +188,11 @@ where
                 let service = self.bind.bind(&addr).map_err(|_| ())?;
 
                 Ok(Async::Ready(Change::Insert(addr, service)))
-            }
+            },
+            // TODO: handle metadata changes by changing the labeling
+            // middleware to hold a `futures-watch::Watch` on the label value,
+            // so it can be updated.
+            Update::ChangeMetadata(_) => unimplemented!(),
```

Comment:
Hmm. To me, I think it makes much more sense for `Update::ChangeMetadata` to be introduced in this PR rather than in #661. However, I understand why you would prefer to avoid `unimplemented!()`, so I'll make that change.

Reply:
Okay, fixed in dfe8a1febb78ed54700aeee775a6d2b1f305831a --- making `DestinationSet::on_change` no-op in the case that would generate a `ChangeMetadata` event makes that function a bit uglier, but there are no panics, and this code is for an interim  state.